### ISSUE TITLE
Inconsistency in vocabulary size of MoE model

### ISSUE THREAD

**Issue by hyhuang00** [facebookresearch/fairseq#4274]
### Before asking:
1. search the issues.
2. search the docs.

#### What is your question?
There seems to be an inconsistency in the vocabulary size of the MoE model. The MoE model, as described in the MoE branch and [appendix of the paper](https://arxiv.org/pdf/2112.10684.pdf), has a vocabulary size of 51200, which is consistent with the pretrained models provided by the MoE branch. However,  [section 3.2 of the paper] suggests the data is actually processed with a vocabulary of 50K subword units, which seems to be the default setting for the training snippet provided [here](https://github.com/pytorch/fairseq/blob/moe/README.moe.md), since the training script prints the following:.

`        (embed_tokens): Embedding(50000, 768, padding_idx=1)`

 I have the following questions:

1. Which vocabulary size is the correct setting, 51200 or 50000? 
2.  To evaluate the pretrained model as suggested in the [README](https://github.com/pytorch/fairseq/blob/moe/README.moe.md), a preprocessed dataset (say, wikitext-103) is needed. However, it currently has a vocabulary size between 50000 and 51200. It seems like this issue can be mitigated via fixing the dictionary (`dictionary.finalize(padding_factor=1024)`), but I'm not sure whether this is the correct way to fix it.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or main): moe
 - PyTorch Version (e.g., 1.0): 1.10.2
 - OS (e.g., Linux): Ubuntu
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): `pip install -v --no-cache-dir --global-option="--cpp_ext" \
    --global-option="--cuda_ext" --global-option="--deprecated_fused_adam" \
    --global-option="--xentropy" --global-option="--fast_multihead_attn" \
    git+git://github.com/NVIDIA/apex.git@e2083df5eb96643c61613b9df48dd4eea6b07690`
 - Python version: 3.9.10
 - CUDA/cuDNN version: 11.3
 - GPU models and configuration: NVIDIA RTX A5000
 - Any other relevant information:

**Comment by jzhou316** [comment]
I have the similar question. The GPT2 dictionary, which is the dictionary being used as described in the MoE paper and can be downloaded from https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/dict.txt, is of size 50260. But the model embeddings have size 51200 from the checkpoints. Can you please confirm:
1. https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/dict.txt is the right dictionary?
2. 51200 is obtained by padding empty entries in the above dictionary in the end, which was never used?

Furthermore, running the MoE LM inference code incurs an error of mismatching dictionary size and the embedding size when directly using the GPT2 bpe dictionary. What is the most convenient way to fix it?
